### PR TITLE
project updated version 

### PR DESCRIPTION
**Pull request by DimQ1**

### CODE REVIEW

File: httprequest.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+const https = require('https');
+const parseJsonResponse = require('./parseJsonData');
+
+function httprequest(path, callback) {
```
Comment (BSiarhei) [inline httprequest.js]: it should return a promise

File: httprequest.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+    const req = https.request(options, (res) => {
+        let data = '';
+        res.on('data', (d) => {
+            data += d;
+        });
+        res.on('end', () => {
+            const parsedData = parseJsonResponse(data);
+
+            return callback(parsedData.jokes, parsedData.pages);
```
Comment (BSiarhei) [inline httprequest.js]: it should return just a data (maybe parsed data)

File: httprequest.js (hunk 1 above)
Comment (BSiarhei) [inline httprequest.js]: Why all name are in camelCase but this one is in lowercase?

File: httprequest.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+const https = require('https');
+
+function httpRequest(path) {
```
Comment (BSiarhei) [inline httprequest.js]: nodejs best practices, not to name variables as core modules

File: httprequest.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        method: 'GET',
+        headers: { Accept: 'application/json' },
+        path: path.pathname + path.search
+    };
+
+    return new Promise((resolve, reject) => {
+        const req = https.request(options,
+            (res) => {
+                let body = '';
+                res.on('data', (chunk) => { body += chunk.toString(); });
```
Comment (BSiarhei) [inline httprequest.js]: without casting

File: askTypeTask.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+            name: 'value',
+            message: 'select type query?',
+            choices: [
+                { title: 'Random', value: 0 },
+                { title: 'Search Tearm', value: 1 },
+                { title: 'Display liderboard joke', value: 2 }
+            ]
+        },
+        {
+            type: prev => prev === 1 ? 'text' : null,
```
Comment (BSiarhei) [inline askTypeTask.js]: bad reduction

File: parseJsonData.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
+function convertJokeToStr(joke) {
+    return `${joke.id}|${joke.joke}|\r\n`;
+}
+
+function parseJsonResponseData(data) {
+    let jokes = '';
+    let pages = 0;
+    const parseData = JSON.parse(data);
+    if ('results' in parseData) {
+        JSON.parse(data).results.forEach((element) => {
```
Comment (BSiarhei) [inline parseJsonData.js]: why do you parse twice

File: saveJoke.js [hunk 7]
```diff
@@ -0,0 +1,13 @@
+const path = require('path');
+const fs = require('fs');
+
+module.exports = function (fileName) {
+    const filePath = path.join(__dirname, fileName);
+
+    return function (textJoke) {
```
Comment (BSiarhei) [inline saveJoke.js]: why do you use Joke in title? and title of the file?

File: leaderJoke.js [hunk 8]
```diff
@@ -0,0 +1,20 @@
+module.exports = (async (jokes) => {
+    const jokeStatistic = jokes.reduce((acc, el) => {
```
Comment (BSiarhei) [inline leaderJoke.js]: bad reduction

File: loadJokeFromFile.js [hunk 9]
```diff
@@ -0,0 +1,29 @@
+const fs = require('fs');
+
+const fsPromises = fs.promises;
+const path = require('path');
+
+function parse(fileText) {
+    const regexp = /(\w+?\|)(.|\r\n)+?(\|\r\n)/g;
```
Comment (BSiarhei) [inline loadJokeFromFile.js:7]: can explain?

File: leaderJoke.js [hunk 10]
```diff
@@ -0,0 +1,20 @@
+module.exports = (async (jokes) => {
```
Comment (BSiarhei) [inline leaderJoke.js]: why file is named as a noun?

File: jokeServise.js [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+                saveJoke(parsedPageData.jokes);
+            }));
+    } else {
+        saveJoke(parsedData.jokes);
+    }
+}
+
+module.exports = {
+    searchJoke: function (searchTerm) {
+        getSearchTermJokeData(searchTerm, saveJoke.toFile);
```
Comment (BSiarhei) [inline jokeServise.js]: return?

File: jokeServise.js [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+    },
+    getRandomJoke: async function () {
+        const response = await httprequest(jokeUrl);
+        const parsedData = parseJsonResponse(response);
+        saveJoke(parsedData.jokes);
```
Comment (BSiarhei) [inline jokeServise.js]: await?

File: loadJokeFromFile.js (hunk 9 above)
Comment (DimQ1) [reply loadJokeFromFile.js:7]: \w - all wards
+  - one or more previous symbols  
? - lazy search
\| - symbol "|", "\" - escape character (экранирующий символ. просто я мог неправильно перевести:) )
(.|\r\n) - any symbol or new carriage return (переход на новую строку)
g - Global search.

Comment (BSiarhei) [reply loadJokeFromFile.js:7]: I asked why do you use regexp but not how it works

**Pull request closed without merging**